base_layer/core/src/base_node/sync/block_sync/synchronizer.rs

```diff
@@ -245,24 +250,36 @@ impl<B: BlockchainBackend + 'static> BlockSynchronizer<B> {
             end_hash: tip_hash.to_vec(),
         };
 
-        let mut block_stream = client.sync_blocks(request).await?;
+        let mut block_stream = match client.sync_blocks(request).await {
+            Ok(stream) => stream,
+            Err(err) => {
+                warn!(target: LOG_TARGET, "RPC request error: {}", err);
+                return Err(BlockSyncError::CommsErrorRetryNextPeer);
```

Comment:
why dont you return the actual error here?
This is more a next logical step than an error.